src/main/java/frc/robot/poseestimator/WPILibOdometryEstimator.java

```diff
@@ -0,0 +1,48 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.Odometry;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import frc.robot.poseestimator.observations.OdometryObservation;
+import frc.robot.subsystems.GBSubsystem;
+
+public class WPILibOdometryEstimator extends GBSubsystem implements IOdometryEstimator {
+
+	private final Odometry<SwerveModulePosition[]> odometryEstimator;
+
+	public WPILibOdometryEstimator(
+		String logPath,
+		SwerveDriveKinematics kinematics,
+		Rotation2d initialGyroAngle,
+		SwerveModulePosition[] initialWheelPositions,
+		Pose2d initialPose
+	) {
+		super(logPath);
+		this.odometryEstimator = new Odometry<>(kinematics, initialGyroAngle, initialWheelPositions, initialPose);
+	}
+
+	@Override
+	public void updateOdometry(OdometryObservation[] odometryObservations) {
+		for (OdometryObservation observation : odometryObservations) {
+			odometryEstimator.update(observation.gyroAngle(), observation.wheelPositions());
```

Comment:
Use the observation timestamp, I am pretty sure u can and WPILib have with time function

Reply:
maybe use the `SwerveDrivePoseEstimator`. it has `updateWithTime()`